ctapipe_io_lst/tests/test_calib.py

```diff
@@ -0,0 +1,154 @@
+import os
+from pathlib import Path
+from traitlets.config import Config
+import numpy as np
+import tables
+import pkg_resources
+import json
+from ctapipe.io.astropy_helpers import h5_table_to_astropy
+
+resource_dir = Path(pkg_resources.resource_filename(
+    'ctapipe_io_lst', 'tests/resources'
+))
+
+test_data = Path(os.getenv('LSTCHAIN_TEST_DATA', 'test_data'))
+test_r0_path = test_data / 'real/R0/20200218/LST-1.1.Run02006.0004.fits.fz'
+test_calib_path = test_data / 'real/calibration/20200218/v05/calibration.Run2006.0000.hdf5'
+test_drs4_pedestal_path = test_data / 'real/calibration/20200218/v05/drs4_pedestal.Run2005.0000.fits'
+test_time_calib_path = test_data / 'real/calibration/20200218/v05/time_calibration.Run2006.0000.hdf5'
+
+
+def test_get_first_capacitor():
+    from ctapipe_io_lst import LSTEventSource
+    from ctapipe_io_lst.calibration import (
+        get_first_capacitor_for_modules,
+        N_GAINS, N_PIXELS_PER_MODULE, N_MODULES,
+    )
+
+    tel_id = 1
+    source = LSTEventSource(test_r0_path)
+    event = next(iter(source))
+
+    first_capacitor_id = event.lst.tel[tel_id].evt.first_capacitor_id
+
+    with tables.open_file(resource_dir / 'first_caps.hdf5', 'r') as f:
+        expected = f.root.first_capacitor_for_modules[:]
+
+    first_caps = get_first_capacitor_for_modules(first_capacitor_id)
+    # we just different shape (N_MODULES, N_GAINS, N_PIXELS_PER_MODULE) before
+    first_caps = first_caps.reshape((N_GAINS, N_MODULES, N_PIXELS_PER_MODULE))
+    first_caps = np.swapaxes(first_caps, 0, 1)
+    assert np.all(first_caps == expected)
+
+
+def test_read_calib_file():
+    from ctapipe_io_lst.calibration import LSTR0Corrections
+
+    mon = LSTR0Corrections._read_calibration_file(test_calib_path)
+    # only one telescope in that file
+    assert mon.tel.keys() == {1, }
+
+
+def test_read_drs4_pedestal_file():
+    from ctapipe_io_lst.calibration import LSTR0Corrections, N_CAPACITORS_PIXEL, N_ROI
+
+    pedestal = LSTR0Corrections._get_drs4_pedestal_data(test_drs4_pedestal_path)
+
+    assert pedestal.shape[-1] == N_CAPACITORS_PIXEL + N_ROI
+    # check circular boundary
+    assert np.all(pedestal[..., :N_ROI] == pedestal[..., N_CAPACITORS_PIXEL:])
+
+    # check offset is applied
+    pedestal_offset = LSTR0Corrections._get_drs4_pedestal_data(
+        test_drs4_pedestal_path, offset=100,
+    )
+    assert np.all((pedestal - pedestal_offset) == 100)
```

Comment:
Oh .. isn't this test repeating the implementation? I think this is frowned upon.

I think the reason why this test has to be written like this, is simply because the function breaks SRP to be more convenient. 
It reads data from a path .. **and** it applies an offset, these things can fail for totally different reasons. 

But ... I guess what I wrote above is not mission critical.

Reply:
It's more about testing that giving the option results in the expected outcome, not that a simple subtraction yields the correct results.

Reply:
> It reads data from a path .. and it applies an offset, these things can fail for totally different reasons.

This is why I test with the same file, one time with offset, one time wihout.

Reply:
Yeah . I guess you have no real choice here.